repos/system_upgrade/el9toel10/actors/sssd/sssdupdate/libraries/sssdupdate.py

```diff
@@ -0,0 +1,81 @@
+import os
+import re
+
+from leapp.exceptions import StopActorExecutionError
+
+
+def _process_knownhosts(line: str) -> str:
+    if re.search(r'^\s*#?\s*ProxyCommand\s+(/usr/bin/)?sss_ssh_knownhostsproxy', line) is not None:
+        # Update the line, leaving intact any # and any --domain/-d parameter
+        line = line.replace('ProxyCommand', 'KnownHostsCommand')
+        line = line.replace('sss_ssh_knownhostsproxy', 'sss_ssh_knownhosts')
+        line = line.replace('--port=', '')
+        line = line.replace('--port', '')
+        line = line.replace('-p=', '')
+        line = line.replace('-p', '')
+        line = line.replace('%p', '')
+        line = line.replace('%h', '%H')
+
+    return line
+
+
+def _process_enable_svc(line: str) -> str:
+    if re.search(r'^\s*#?\s*services\s*=', line) is not None:
+        if re.search(r'=\s*(.+,)?\s*ssh\s*(,.+)?\s*$', line) is None:
+            line = line.rstrip()
+            line += (',' if line[-1] != '=' else '') + 'ssh\n'
+
+    return line
+
+
+def _update_file(filename, process_function):
+    newname = filename + '.new'
+    oldname = filename + '.old'
+    try:
+        with open(filename, 'r') as input_file, open(newname, 'x') as output_file:
```

Comment:
I would just rewrite it (@MichalHe  good point!).

```suggestion
        with open(filename, 'r') as input_file, open(newname, 'w') as output_file:
```